File: app/backend/src/frameworks_and_drivers/persistence/entities/product_entity.rs
use sqlx::FromRow;

/// データベースエンティティ - 製品基本情報
/// Clean Architecture: Frameworks & Drivers層のデータ表現
#[derive(Debug, FromRow)]
pub struct ProductEntity {
    pub id: String,                        // TEXT型 (UUID)
    pub name: String,
    pub description: String,
    pub category_id: String,               // TEXT型 (UUID)
    pub is_best_seller: bool,
    pub is_quick_ship: bool,
    pub is_active: bool,                   // DBの available と同等
    pub created_at: String,
    pub updated_at: String,
}

File: app/backend-ts/src/infrastructure/database/entities/product.entity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  OneToMany,
  PrimaryColumn,
  UpdateDateColumn,
} from 'typeorm';

import { CategoryEntity } from './category.entity';
import { ProductImageEntity } from './product-image.entity';
import { SkuEntity } from './sku.entity';

@Entity('products')
export class ProductEntity {
  @PrimaryColumn('text')
  id: string;

  @Column('text')
  name: string;

  @Column('text')
  description: string;

  @Column('text')
  category_id: string;

  @Column('boolean', { default: false })
  is_best_seller: boolean;

  @Column('boolean', { default: false })
  is_quick_ship: boolean;

  @Column('boolean', { default: true })
  is_available: boolean;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;

  @OneToMany(() => SkuEntity, (sku) => sku.product)
  skus: SkuEntity[];

  @ManyToOne(() => CategoryEntity)
  @JoinColumn({ name: 'category_id' })
  category: CategoryEntity;

  @OneToMany(() => ProductImageEntity, (image) => image.product)
  images: ProductImageEntity[];
}
